src/domains/ioweb-app/06_function_ioweb_profile.tf

```
@@ -316,3 +316,85 @@ resource "azurerm_monitor_autoscale_setting" "function_ioweb_profile" {
     }
   }
 }
+
+// ----------------------------------------------------
+// Alerts
+// ----------------------------------------------------
+resource "azurerm_monitor_scheduled_query_rules_alert_v2" "alert_too_much_invalid_codes_on_unlock" {
+  enabled                 = true
+  name                    = "[${upper(var.domain)} | ${module.function_ioweb_profile.name}] Unexpected number of invalid codes to unlock endpoint"
+  resource_group_name     = azurerm_resource_group.ioweb_profile_rg.name
+  scopes                  = [module.function_ioweb_profile.id]
+  description             = "Too many invalid codes submitted to IO-WEB profile unlock functionality"
+  severity                = 1
+  auto_mitigation_enabled = false
+  location                = azurerm_resource_group.ioweb_profile_rg.location
+
+  // check once every minute(evaluation_frequency)
+  // on the last minute of data(window_duration)
+  evaluation_frequency = "PT1M"
+  window_duration      = "PT1M"
+
+  criteria {
+    query                   = <<-QUERY
+requests
+| where cloud_RoleName == "${module.function_ioweb_profile.name}"
+| where url contains "unlock-session"
```

We can optimize the query using `==` instead `contains`. [ref](https://learn.microsoft.com/it-it/azure/data-explorer/kusto/query/datatypes-string-operators#performance-tips)
```suggestion
| where name == "UnlockSession"
```